chromium_src/components/sync_device_info/device_info_sync_bridge_unittest.cc

```diff
@@ -84,5 +114,46 @@ TEST_F(DeviceInfoSyncBridgeTest, RemoteDelete) {
   EXPECT_THAT(ReadAllFromStore(), UnorderedElementsAre(Pair(kLocalGuid, _)));
 }
 
+TEST_F(DeviceInfoSyncBridgeTest, LocalDeleteWhenOffline) {
+  InitializeAndMergeInitialData(SyncMode::kFull);
+  ASSERT_EQ(1u, bridge()->GetAllDeviceInfo().size());
+  ASSERT_EQ(1, change_count());
+  ASSERT_FALSE(ReadAllFromStore().empty());
+
+  const DeviceInfoSpecifics specifics = CreateSpecifics(1, base::Time::Now());
+  auto error = bridge()->ApplySyncChanges(bridge()->CreateMetadataChangeList(),
+                                          EntityAddList({specifics}));
+
+  ASSERT_FALSE(error);
+  ASSERT_EQ(2u, bridge()->GetAllDeviceInfo().size());
+  ASSERT_EQ(2, change_count());
+
+  const std::string kLocalGuid = CacheGuidForSuffix(kLocalSuffix);
+  ON_CALL(*processor(), IsEntityUnsynced(kLocalGuid))
+      .WillByDefault(Return(true));
+  EXPECT_CALL(*processor(), IsEntityUnsynced).Times(5);
+  EXPECT_CALL(*processor(), Delete(kLocalGuid, _)).Times(1);
+
+  bool deleted_device_info_sent = false;
+  base::RunLoop loop;
+  bridge()->DeleteDeviceInfo(
+      kLocalGuid, base::BindOnce(
+                      [](base::RunLoop* loop, bool* deleted_device_info_sent) {
+                        *deleted_device_info_sent = true;
+                        loop->Quit();
+                      },
+                      &loop, &deleted_device_info_sent));
+
+  loop.Run();
+
+  EXPECT_EQ(5, bridge()->GetDeviceDeleteAttemptsForTest());
```

Comment:
I was thinking that we can drop this when we started to count `IsEntityUnsynced` calls. wdyt? it would be a little cleaner.

Reply:
thanks, with that was possible to remove both `GetDeviceDeleteAttemptsForTest` and `device_delete_attempts_for_test_`